app/controllers/ops_controller/ops_rbac.rb

```diff
@@ -349,9 +349,17 @@ def rbac_tenants_list
   def rbac_tenant_delete
     assert_privileges("rbac_tenant_delete")
     tenants = []
+    projects = []
     if !params[:id] # showing a tenants list
-      ids = find_checked_items.collect { |r| from_cid(r.split("-").last) }
-      tenants = Tenant.find_all_by_id(ids)
+      ids = find_checked_items
+      recs_to_delete = Tenant.find_all_by_id(ids)
```

Comment:
try and avoid `find_all_by_...`, use `Tenant.where(:id => ids)` instead. If you need to make sure it loads from the db, tack on `to_a`.

May want to check out [`partition`](http://ruby-doc.org/core-2.3.0/Enumerable.html#method-i-partition), although I'm not sure you need to partition at all

```ruby
tenants, projects = recs_to_delete.partition(&:divisible)
```